src/js/letters/components/Address.jsx

```diff
@@ -79,96 +48,113 @@ class Address extends React.Component {
       });
     }
 
-    return states;
+    return statesList;
+  }
+
+  isMilitaryCity = (city) => {
+    const lowerCity = city.toLowerCase().trim();
+
+    return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
   }
 
   render() {
-    const addressType = this.props.address.type;
+    const errorMessages = this.props.errorMessages;
 
     // Our hard-coded list of countries has the value for the U.S. as
     // 'USA', but the value sent to us from EVSS is 'US'. This will cause
     // the field to not populate correctly.
     // This will be changed once we pull the real country list from the
     // address endpoint.
     let selectedCountry;
-    if (this.props.address.country === undefined && addressType === 'MILITARY') {
+    if (this.props.address.country === undefined && this.props.address.type === ADDRESS_TYPES.military) {
```

Comment:
Are there cases where the country could be undefined and the address type is something other than Military? If so, looks like we're defaulting to setting selectedCountry to the country in the address prop. I take it this maps to an empty country dropdown on the front end?

Do we need the `undefined` check?

Reply:
Hmmm...I'm actually no sure; I didn't write that bit. Lemme look!

Reply:
My guess is when we get an address from the api which has a type of `MILITARY`, but no country. We want to select 'USA' on the front end so we can do everything we need to...

@annekainicUSDS to verify.

Reply:
@markgreenburg @cvalarida according to the table in this ticket (https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4110), I believe it is possible for `country` to be `undefined`, and for the `addressType` to be `domestic`. In the table it states that `countryName` is not required for domestic addresses. `country` is only required for international addresses. So in order to guarantee the address is military, I think we need to do both of these checks.

However, since we're now pulling the countries from the API, I'm not sure any of this code is needed anymore? I can address that in a separate PR once this is merged.